File: ios-im-ui/Sources/FlareIMUI/Models/ConversationSummary.swift
import Foundation

/// Conversation kind — spec union `'single' | 'group' | 'ai'`.
public enum FlareConversationKind: Sendable {
    case single, group, ai
}

/// Neutral summary of a conversation for the details/settings panel — the
/// spec's `Conversation` type as consumed by ``ConversationDetailsView``.
public struct FlareConversationSummary: Sendable {
    public let id: String
    public let title: String
    public let avatarURL: String?
    public let kind: FlareConversationKind
    public let memberCount: Int?
    public let muted: Bool
    public let pinned: Bool
    public let archived: Bool

    public init(
        id: String,
        title: String,
        avatarURL: String? = nil,
        kind: FlareConversationKind = .single,
        memberCount: Int? = nil,
        muted: Bool = false,
        pinned: Bool = false,
        archived: Bool = false
    ) {
        self.id = id
        self.title = title
        self.avatarURL = avatarURL
        self.kind = kind
        self.memberCount = memberCount
        self.muted = muted
        self.pinned = pinned
        self.archived = archived
    }
}

/// A selectable contact/directory entry for ``StartConversationView``.
public struct FlareContactOption: Identifiable, Sendable {
    public let id: String
    public let name: String
    public let avatarURL: String?
    public let subtitle: String?

    public init(id: String, name: String, avatarURL: String? = nil, subtitle: String? = nil) {
        self.id = id
        self.name = name
        self.avatarURL = avatarURL
        self.subtitle = subtitle
    }
}

File: packages/flutter-im-ui/lib/src/models/conversation_summary.dart
import '../components/flare_message_bubble.dart' show FlareConversationKind;

/// Neutral summary of a conversation for the details/settings panel — the
/// spec's `Conversation` type as consumed by `FlareConversationDetails`.
class FlareConversationSummary {
  const FlareConversationSummary({
    required this.id,
    required this.title,
    this.avatarUrl,
    this.kind = FlareConversationKind.single,
    this.memberCount,
    this.muted = false,
    this.pinned = false,
    this.archived = false,
  });

  final String id;
  final String title;
  final String? avatarUrl;
  final FlareConversationKind kind;
  final int? memberCount;
  final bool muted;
  final bool pinned;
  final bool archived;
}
